crates/ruff/src/rule_selector.rs

```diff
@@ -61,11 +66,20 @@ impl FromStr for RuleSelector {
                     return Ok(Self::Linter(linter));
                 }
 
-                Ok(Self::Prefix {
-                    prefix: RuleCodePrefix::parse(&linter, code)
-                        .map_err(|_| ParseError::Unknown(s.to_string()))?,
-                    redirected_from,
-                })
+                // Does the selector select a single rule?
+                let prefix = RuleCodePrefix::parse(&linter, code)
+                    .map_err(|_| ParseError::Unknown(s.to_string()))?;
+                if prefix.clone().rules().count() == 1 {
```

Comment:
This isn't sufficiently robust, it can be refined... (E.g., this would match `--select CPY` since there's only one rule in that category).

Reply:
(Fixed.)